Source/WebCore/Modules/paymentrequest/MerchantValidationEvent.cpp

```diff
@@ -43,7 +43,7 @@ Ref<MerchantValidationEvent> MerchantValidationEvent::create(const AtomString& t
 
 ExceptionOr<Ref<MerchantValidationEvent>> MerchantValidationEvent::create(Document& document, const AtomString& type, Init&& eventInit)
 {
-    URL validationURL { document.url(), eventInit.validationURL };
+    auto validationURL = document.completeURL(eventInit.validationURL);
```

Comment:
To ensure UTF-8 you need to pass ForceUTF8::Yes. Thankfully you had included the need in your commit message as I had forgotten about it.

Reply:
```auto validationURL = document.completeURL(eventInit.validationURL, ForceUTF8::Yes);```

Or I have to add `ScriptExecutionContext` as well?

Reply:
Yeah maybe.

Reply:
Thanks! I was under impression that completeURL() would implicitly will have UTF8 support. Thanks for correcting myself and learnt new stuff as well.

Reply:
Done in [b0a06fa](https://github.com/WebKit/WebKit/commit/b0a06fa23c984a51c24687f122f9ed1551f88014). Confirmed that it compiles locally.